src/routes/source.js

```diff
@@ -23,8 +24,9 @@ async function invalidateCollab(api, url, env) {
   await env.dacollab.fetch(invURL);
 }
 
-export async function deleteSource({ env, daCtx }) {
-  return /* await */ deleteObjects(env, daCtx);
+export async function deleteSource({ req, env, daCtx }) {
+  const details = await deleteHelper(req);
+  return /* await */ deleteObjects(env, daCtx, details);
```

Comment:
Why is the await commented out instead of deleted?

Reply:
I do not know. I left it there as it was from the original variation of this work.

Reply:
I did that 😄 
The reason is that I find it problematic that all of a sudden the linter _wants_ you to lose the await on an async function if its called on `return`. If you change the function later and want to add some statements before returning you might forget to add the `await` at that point.
So I just added the await in a comment as a reminder. 

But if you don't like it feel free to remove 😄